internal/access/group.go

```diff
@@ -30,13 +33,29 @@ func isUserInGroup(c *gin.Context, requestedResourceID uid.ID) (bool, error) {
 	return false, nil
 }
 
-func ListGroups(c *gin.Context, name string) ([]models.Group, error) {
+func ListGroups(c *gin.Context, name string, userID uid.ID) ([]models.Group, error) {
+	var selectors []data.SelectorFunc
+	if name != "" {
+		selectors = append(selectors, data.ByName(name))
+	}
+	if userID != 0 {
+		selectors = append(selectors, data.WhereGroupIncludesUser(userID))
+	}
+
 	db, err := RequireInfraRole(c, models.InfraAdminRole, models.InfraViewRole, models.InfraConnectorRole)
-	if err != nil {
-		return nil, err
+	if err == nil {
+		return data.ListGroups(db, selectors...)
+	}
+
+	if errors.Is(err, internal.ErrForbidden) {
+		db, err := hasAuthorization(c, userID, isIdentitySelf)
+		if err != nil {
+			return nil, err
+		}
+		return data.ListGroups(db, selectors...)
```

Comment:
We need to add a selector to filter this by the user's groups. they don't automatically get access to do whatever query they are asking for